core/arch/arm/mm/core_mmu_v7.c

```diff
@@ -671,8 +671,10 @@ static void map_memarea(struct tee_mmap_region *mm, uint32_t *ttb)
 	 * TODO: support mapping devices at a virtual address which isn't
 	 * the same as the physical address.
 	 */
-	if (mm->va < (NUM_UL1_ENTRIES * SECTION_SIZE))
-		panic("va conflicts with user ta address");
+	if (ttb == (void *)core_mmu_get_main_ttb_va()) {
```

Comment:
ok, I understand. I think this deserves an explicit comment.

The map_memarea() routine calls map_mem_page() below. This routine allocates level2 tables in the OP-TEE generic nozi area, hence in the main external RAM when paging is disable. This will conflict with your needs.

I feel a bit afraid using this routines in a scope they were not design for. I feel they will need deeper rework. Could you give us a more global picture of the OP-TEE SW suspend/resume architecture, to understand your needs.